src/dart/codegen.rs

```diff
@@ -1,129 +1,226 @@
         match *item {
-            Item::ComponentDef(ref name, ref fields, ref instance) => {
-                self.write_str("class ");
-                self.write_ident(name);
-                self.write_str(" extends StatelessWidget {\n");
+            Item::ComponentDef(name, ref fields, ref instance) => {
+                let params;
+                let body;
+                let superclass;
+                let mut class_members = vec![];
+                superclass = Some(ast::Type::simple_path("StatelessWidget"));
                 if !fields.is_empty() {
-                    self.write_str("const ");
-                    self.write_ident(name);
-                    self.write_str("({ Key key");
+                    let mut normal = vec![];
+                    let mut default = vec![];
+                    normal.push(ast::ArgDef::build_param(
+                        ast::Type::simple_path("Key"),
+                        false,
+                        "key",
+                    ));
+
                     for field in fields {
-                        self.write_str(", this.");
-                        self.write_ident(&field.name);
+                        let mut var_ty = Node::new(ast::Type::Infer);
+                        if let Some(ref ty) = field.ty {
+                            var_ty = self.codegen_type(ty);
+                        }
+                        let arg = ast::ArgDef {
+                            metadata: vec![],
+                            covariant: false,
+                            ty: ast::VarType {
+                                fcv: ast::FinalConstVar::Var,
+                                ty: var_ty,
+                            },
+                            field: true,
+                            name: field.name,
+                        };
                         if let Some(ref value) = field.default {
-                            self.write_str(" = ");
-                            self.codegen_expr(value);
+                            default.push(ast::OptionalArgDef {
+                                arg,
+                                default: Some(self.codegen_expr(value)),
+                            });
+                        } else {
+                            normal.push(arg);
                         }
                     }
-                    self.write_str(" }) : super(key: key);\n");
-                    self.codegen_field_defs(fields);
+                    params = ast::FnSig {
+                        return_type: Node::new(ast::Type::Infer),
+                        required: normal,
+                        optional: default,
+                        optional_kind: ast::OptionalArgKind::Named,
+                        async: false,
+                        generator: false,
+                    };
+                    let mut qualif = vec![];
+                    qualif.push(ast::MethodQualifiers::Const);
```

Comment:
This can be just `vec![ast::MethodQualifiers::Const]` - and used only once, 3 lines below.